kalite/distributed/static/js/distributed/user/views.js

```diff
@@ -314,4 +314,64 @@ window.UserView = BaseView.extend({
     logout: function() {
         this.model.logout();
     }
-});
\ No newline at end of file
+});
+
+/* This view toggles which navbar items are displayed to each type of user */ 
+window.ToggleNavbarView = BaseView.extend ({
+
+    template: HB.template("user/navigation"),
+
+    initialize: function() {
+
+        _.bindAll(this);        
+        this.listenTo(this.model, "change", this.render);
+        $("topnav").append(this.template());
+
+    },
+
+    render: function() {
+        
+        this.facility_id = {"facility_id": this.model.get("facility_id")};
+        this.$el.html(this.template(this.facility_id));
+        
+        this.userView = new UserView({ model: this.model, el: "#topnav" });
+        $("#topnav").prepend(new AutoCompleteView().$el);
+        
+        // GUEST
+        if ( !this.model.get("is_logged_in") ) {
+            this.$(".guest").removeClass("display-none");
+        }
+        // ALL LOGGED IN
+        if ( this.model.get("is_logged_in") ) {
+            this.$(".logged-in").removeClass("display-none");
+        }
+        // ALL USERS - slight lag in tabs appearing, seems unnecessary but ensures that all tabs appear at the same time
+        if ( this.model.get("is_logged_in") || !this.model.get("is_logged_in") ) {
+            this.$(".learn-tab").removeClass("display-none");
+        }
+        // STUDENTS
+        if ( this.model.get("is_logged_in") && !this.model.get("is_admin") ) {
+            this.$(".student").removeClass("display-none");
+        }
+        // TEACHER
+        if ( !this.model.get("is_django_user") && this.model.get("is_admin") ) {
+            this.$(".teacher").removeClass("display-none");
+        }
+        // ADMINS
+        if ( this.model.get("is_django_user") ) {
+            this.$(".admin").removeClass("display-none");
+        }
+
+        // ACTIVATES NAV TAB ACCORDING TO CURRENT PAGE
```

Comment:
I think this should be an attribute on the status model (and then could be handled in the template as above).

Doing it this way means you won't have to worry about a race condition on rendering - whereby if this renders before your ```var onTeachPage``` is set, or whichever, then you might end up with no active tab.